File: Day 1/by.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter words separated by space: ");
        String[] words = scanner.nextLine().split(" ");
        System.out.println("Enter a character: ");
        char c = scanner.next().charAt(0);

        int count = 0;
        Map<String, List<Integer>> positions = new HashMap<>();
        for (String word : words) {
            List<Integer> indices = new ArrayList<>();
            for (int i = 0; i < word.length(); i++) {
                if (word.charAt(i) == c) {
                    count++;
                    indices.add(i);
                }
            }
            positions.put(word, indices);
        }

        System.out.println("'" + c + "' appears " + count + " times.");
        System.out.println("Positions in each word: " + positions);
    }
}

File: Day 1/by.php
<?php
function count_char($words, $char) {
    $count = 0;
    $positions = array();
    foreach ($words as $word) {
        $count += substr_count($word, $char);
        $pos = array();
        for ($i = 0; $i < strlen($word); $i++) {
            if ($word[$i] == $char) {
                array_push($pos, $i);
            }
        }
        $positions[$word] = $pos;
    }
    return array($count, $positions);
}

echo "Enter words separated by space: ";
$words = explode(" ", trim(fgets(STDIN)));
echo "Enter a character: ";
$char = trim(fgets(STDIN));
list($count, $positions) = count_char($words, $char);
echo "'$char' appears $count times.\n";
echo "Positions in each word:\n";
print_r($positions);
?>
